### PR TITLE
Lg/candid users guide

### PR DESCRIPTION
**Pull request by lsgunnlsgunn**
Split out into separate sections for adding to the SDK website, fix some links, add some Rust content and links.

### CODE REVIEW

File: docs/modules/candid-guide/pages/candid-ref.adoc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+= Reference
+
+This section provides detailed reference information about Candid supported types and links to Candid tools and libraries for currently supported host languages.
+
+* link:candid-types{outfilesuffix}[Supported types]
+* link:candid-tools{outfilesuffix}[Tools and libraries]
```
Comment (rossberg) [inline docs/modules/candid-guide/pages/candid-ref.adoc:6]: Sorry, I still think that putting tools under "Reference" makes no sense. A language reference is a detailed description of a language, not its tool chain. Can we change this?

File: docs/modules/candid-guide/pages/candid-types.adoc [hunk 2]
```diff
@@ -0,0 +1,828 @@
+= Supported types
```
Comment (rossberg) [inline docs/modules/candid-guide/pages/candid-types.adoc:1]: This doc is the real language reference.

File: docs/modules/candid-guide/pages/candid-ref.adoc (hunk 1 above)
Comment (lsgunnlsgunn) [reply docs/modules/candid-guide/pages/candid-ref.adoc:6]: I understand your point. I think you might be interpreting "reference" here more literally than was intended. I was using it as a classification umbrella for the type of information it contains rather than to specifically mean a language reference section. While I can move it or remove it, I'd like the flexibility to use this section for other topics that fall under the broader "reference" category and not strictly to mean "language reference".

**Review by nomeata — APPROVED** [review]

File: docs/modules/candid-guide/pages/candid-ref.adoc (hunk 1 above)
Comment (rossberg) [reply docs/modules/candid-guide/pages/candid-ref.adoc:6]: Well, a section "reference" in a language guide naturally is the language reference. ;) Sorry for being pedantic, but the term has a pretty firmly established meaning in the context of language docs, which I'd rather stay consistent with.

File: docs/modules/candid-guide/pages/candid-howto.adoc [hunk 3]
```diff
@@ -81,12 +83,22 @@ You could configure the project `+dfx.json+` file to set the path to the service
...
+In addition to the Candid implementations for Motoko, Rust, and JavaScript, there are community-supported Candid libraries for the following host languages:
+
+* link:https://hackage.haskell.org/package/candid[Haskell]
+* link:https://github.com/chenyan2002/ic-elm/[Elm]
+* link:https://github.com/seniorjoinu/candid-kt[Kotlin]
+
+If you want to create a Candid implementation to support a language or tool for which an implementation is not currently available, you should consult the https://github.com/dfinity/candid/blob/master/spec/Candid.md[Candid specification].
 
-Additionally, the official https://github.com/dfinity/candid/tree/master/test[Candid test data] is useful to make sure your implementation is compatible with Candid, even in slightly more obscure corner cases.
+If you add a Candid implementation for a new language or toll, you can use the official https://github.com/dfinity/candid/tree/master/test[Candid test data] to test and verify that your implementation is compatible with Candid, even in slightly more obscure corner cases.
```
Comment (nomeata) [inline docs/modules/candid-guide/pages/candid-howto.adoc:104]: Or toll?
